frontend/src/components/pages/RoutePlanningPage.tsx

```diff
@@ -5,8 +5,19 @@ import { useSearchParams } from "react-router-dom";
 import { useIsPhone } from "@/components/context/PhoneContext.tsx";
 import { createOrUpdateRoute } from "@/api/map/route_store.ts";
 import { Button } from "@/components/ui/button.tsx";
+import type { Point, RouteSegment } from "@/maps/manage/structure.ts";
+import { useEffect } from "react";
 
-const MapWrapper = () => {
+const MapWrapper = ({
+  onRouteDataChange,
+  initialData,
+}: {
+  onRouteDataChange: (data: {
```

Comment:
Actually, instead of needing an onRouteDataChange function to reach RoutePlanningPage, can we just bind useRouteOperations from operations.ts? That is where we are really storing data, and the current solution is really data ineffective. We have stored two copies of the same data.